simleague/simset.py

```diff
@@ -180,10 +180,35 @@ async def halftimebreak(self, ctx, time: int = 1):
     @simset.command()
     async def resultchannel(self, ctx, channel: discord.TextChannel):
         """Add a channel for automatic result posting."""
-        async with self.config.guild(ctx.guild).resultchannel() as result:
-            result.append(channel.id)
+        channels = await self.config.guild(ctx.guild).resultchannel()
+        for c in channels:
+            if c == channel.id:
+                await ctx.send("Results are already posted in this channel")
+                return
+
+        channels.append(channel.id)
+        await self.config.guild(ctx.guild).resultchannel.set(channels)
```

Comment:
This can be made simpler using a context manager

```py
async with self.config.guild(ctx.guild).resultchannel() as channels:
    if channel in channels:
		await ctx.send("Results are already posted in this channel")
        return
    channels.append(channel.id)```

Reply:
Done